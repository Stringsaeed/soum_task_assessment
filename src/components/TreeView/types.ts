export interface TreeViewItem {
  id: string;
  name: string;
  data: TreeViewItem[];
}

export interface TreeViewProps {
  data: TreeViewItem[];
  selectedValue?: string[];
}
