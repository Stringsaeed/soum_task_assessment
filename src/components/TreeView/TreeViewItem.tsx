import { Minus, Plus } from 'lucide-react-native';
import { PropsWithChildren, useState } from 'react';
import { Pressable, StyleSheet, Text } from 'react-native';
import Animated, { LinearTransition, FadeInDown, FadeOutUp } from 'react-native-reanimated';

import { TreeViewItem as ITreeViewItem } from './types';

import Box from '@/components/Box';
import CheckBox from '@/components/CheckBox';
import { useSelection } from '@/contexts';

type Props = PropsWithChildren<{ item: ITreeViewItem }>;

const AnimatedPressable = Animated.createAnimatedComponent(Pressable);

export default function TreeViewItem({ item, children }: Props) {
  const isExpanderItem = item.data.length;
  const { handleSelect, selectedValue } = useSelection();
  const isSelected = selectedValue.includes(item.id);
  const [isExpanded, setIsExpanded] = useState(isSelected);

  const onPress = () => {
    if (isExpanderItem) {
      setIsExpanded(!isExpanded);
    } else {
      handleSelect(item.id);
    }
  };

  return (
    <AnimatedPressable
      testID="TreeViewItem"
      layout={LinearTransition}
      onPress={onPress}
      style={styles.container}>
      {!isExpanderItem && <CheckBox status={isSelected ? 'checked' : 'unchecked'} />}
      {!!isExpanderItem && (
        <Box size={20} factor={0.8} color="beige" foregroundColor="black">
          {isExpanded ? <Minus size={16} color="black" /> : <Plus size={16} color="black" />}
        </Box>
      )}
      <Animated.View
        testID="TreeViewDataRoot"
        layout={LinearTransition}
        style={styles.nameDataContainer}>
        <Text style={[styles.name, isSelected && styles.nameSelected]}>{item.name}</Text>
        {isExpanded && (
          <Animated.View
            testID="TreeViewItemExpanded"
            entering={FadeInDown}
            exiting={FadeOutUp}
            layout={LinearTransition}
            style={styles.expanded}>
            {!!isExpanderItem && (
              <TreeViewItem item={{ ...item, name: `All in ${item.name}`, data: [] }} />
            )}
            {children}
          </Animated.View>
        )}
      </Animated.View>
    </AnimatedPressable>
  );
}

const styles = StyleSheet.create({
  container: { flexDirection: 'row', gap: 4 },
  name: { fontSize: 16, lineHeight: 20 },
  nameSelected: { fontWeight: '600' },
  nameDataContainer: {
    overflow: 'hidden',
    flex: 1,
    gap: 8,
  },
  expanded: { gap: 8, overflow: 'hidden' },
});
