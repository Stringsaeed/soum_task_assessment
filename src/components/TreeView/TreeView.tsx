import React from 'react';
import Animated, { LinearTransition } from 'react-native-reanimated';

import TreeViewItem from './TreeViewItem';
import { TreeViewProps } from './types';

export default function TreeView({ data }: TreeViewProps) {
  return (
    <Animated.View layout={LinearTransition} testID="TreeView" style={{ gap: 8 }}>
      {data.map((item) => (
        <TreeViewItem item={item} key={item.id}>
          <TreeView data={item.data} />
        </TreeViewItem>
      ))}
    </Animated.View>
  );
}
