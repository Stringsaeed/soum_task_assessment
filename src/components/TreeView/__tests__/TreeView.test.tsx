import { fireEvent, render, screen, waitFor } from '@testing-library/react-native';

import TreeView from '../TreeView';

import { SelectionProvider } from '@/contexts';

const data = [
  {
    id: 'phone_category',
    name: 'Phone Category',
    data: [
      {
        id: 'apple',
        name: 'Apple',
        data: [
          {
            id: 'iphone',
            name: 'iPhone',
            data: [
              {
                id: 'iphone_12',
                name: 'iPhone 12',
                data: [],
              },
              {
                id: 'iphone_11',
                name: 'iPhone 11',
                data: [],
              },
            ],
          },
        ],
      },
      {
        id: 'samsung',
        name: 'Samsung',
        data: [
          {
            id: 'galaxy',
            name: 'Galaxy',
            data: [
              {
                id: 'galaxy_s21',
                name: 'Galaxy S21',
                data: [],
              },
              {
                id: 'galaxy_s20',
                name: 'Galaxy S20',
                data: [],
              },
            ],
          },
        ],
      },
    ],
  },
  {
    id: 'laptop_category',
    name: 'Laptop Category',
    data: [
      {
        id: 'dell',
        name: 'Dell',
        data: [
          {
            id: 'xps',
            name: 'XPS',
            data: [],
          },
          {
            id: 'inspiron',
            name: 'Inspiron',
            data: [],
          },
        ],
      },
      {
        id: 'macbook',
        name: 'MacBook',
        data: [
          {
            id: 'macbook_air',
            name: 'MacBook Air',
            data: [],
          },
          {
            id: 'macbook_pro',
            name: 'MacBook Pro',
            data: [],
          },
        ],
      },
    ],
  },
  {
    id: 'tablet_category',
    name: 'Tablet Category',
    data: [
      {
        id: 'tablet_samsung',
        name: 'Samsung',
        data: [
          {
            id: 'galaxy_tab',
            name: 'Galaxy Tab',
            data: [],
          },
        ],
      },
    ],
  },
  {
    id: 'watch_category',
    name: 'Watch Category',
    data: [
      {
        id: 'apple_watch',
        name: 'Apple Watch',
        data: [
          {
            id: 'apple_watch_series_7',
            name: 'Apple Watch Series 7',
            data: [],
          },
          {
            id: 'apple_watch_series_6',
            name: 'Apple Watch Series 6',
            data: [],
          },
        ],
      },
    ],
  },
];

jest.retryTimes(3);

describe('TreeView', () => {
  it('should render', () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue = ['apple', 'iphone', 'iphone_12'];
    render(<TreeView data={data} />, {
      wrapper: ({ children }) => (
        <SelectionProvider
          selectedValue={mockedSelectedValue}
          onSelectedValueChange={mockedOnSelectedValueChange}>
          {children}
        </SelectionProvider>
      ),
    });

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should render all parent data list', () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue: string[] = [];
    render(<TreeView data={data} />, {
      wrapper: ({ children }) => (
        <SelectionProvider
          selectedValue={mockedSelectedValue}
          onSelectedValueChange={mockedOnSelectedValueChange}>
          {children}
        </SelectionProvider>
      ),
    });

    expect(screen.queryAllByTestId('TreeViewItem').length).toBe(data.length);
  });

  it('should render data with expended state for selection', async () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue = ['phone_category', 'apple', 'iphone', 'iphone_12'];
    render(
      <SelectionProvider
        selectedValue={mockedSelectedValue}
        onSelectedValueChange={mockedOnSelectedValueChange}>
        <TreeView data={data} />
      </SelectionProvider>,
    );

    expect(screen.queryAllByTestId('TreeViewItem').length).toBe(12);
    await screen.findAllByTestId('TreeViewItemExpanded');
  });

  it('should render data with selected state', () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue = ['phone_category', 'apple', 'iphone', 'iphone_12'];
    render(
      <SelectionProvider
        selectedValue={mockedSelectedValue}
        onSelectedValueChange={mockedOnSelectedValueChange}>
        <TreeView data={data} />
      </SelectionProvider>,
    );

    expect(screen.queryAllByTestId(`CheckBox-checked`).length).toBe(4);
  });

  it('should make the parent status checked when all children are selected', () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue = ['phone_category', 'apple', 'iphone', 'iphone_12', 'iphone_11'];
    render(
      <SelectionProvider
        selectedValue={mockedSelectedValue}
        onSelectedValueChange={mockedOnSelectedValueChange}>
        <TreeView data={data} />
      </SelectionProvider>,
    );

    expect(screen.queryAllByTestId(`CheckBox-checked`).length).toBe(5);
  });

  it('should handle onPress for expender item', () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue: string[] = [];
    render(
      <SelectionProvider
        selectedValue={mockedSelectedValue}
        onSelectedValueChange={mockedOnSelectedValueChange}>
        <TreeView data={data} />
      </SelectionProvider>,
    );

    const expandedTreeViewItem = screen.queryAllByTestId('TreeViewItemExpanded');
    const phoneCategoryItemView = screen.getAllByTestId('TreeViewItem')[0];
    fireEvent.press(phoneCategoryItemView);
    const expendedTreeViewItemAfterPress = screen.queryAllByTestId('TreeViewItemExpanded');
    expect(expendedTreeViewItemAfterPress.length).toBe(expandedTreeViewItem.length + 1);
  });

  it('should handle onPress for selection item', async () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue: string[] = [];
    render(
      <SelectionProvider
        selectedValue={mockedSelectedValue}
        onSelectedValueChange={mockedOnSelectedValueChange}>
        <TreeView data={data} />
      </SelectionProvider>,
    );

    const phoneCategoryItemView = screen.getAllByTestId('TreeViewItem')[0];
    fireEvent.press(phoneCategoryItemView);
    await waitFor(() => screen.findByTestId('TreeViewItemExpanded'));
    fireEvent.press(screen.getAllByTestId('TreeViewItem')[1]);
    expect(mockedOnSelectedValueChange).toHaveBeenCalledWith(['phone_category']);
  });

  it('should handle expand if any children is selected', () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue = ['phone_category', 'apple', 'iphone', 'iphone_12'];
    render(
      <SelectionProvider
        selectedValue={mockedSelectedValue}
        onSelectedValueChange={mockedOnSelectedValueChange}>
        <TreeView data={data} />
      </SelectionProvider>,
    );

    expect(screen.queryAllByTestId('TreeViewItemExpanded').length).toBe(7);
    expect(screen.queryAllByTestId('CheckBox-checked').length).toBe(mockedSelectedValue.length);
  });

  it('should set expand if selected after press', async () => {
    const mockedOnSelectedValueChange = jest.fn();
    const mockedSelectedValue: string[] = [];
    render(
      <SelectionProvider
        selectedValue={mockedSelectedValue}
        onSelectedValueChange={mockedOnSelectedValueChange}>
        <TreeView data={data} />
      </SelectionProvider>,
    );

    expect(screen.queryAllByTestId('TreeViewItemExpanded').length).toBe(0);
    await fireEvent.press(screen.getAllByTestId('TreeViewItem')[0]);
    expect(screen.queryAllByTestId('TreeViewItemExpanded').length).toBe(1);
  });
});
