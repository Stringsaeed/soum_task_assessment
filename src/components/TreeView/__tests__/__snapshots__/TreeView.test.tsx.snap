// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeView should render 1`] = `
<View
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  layout={[Function]}
  style={
    {
      "gap": 8,
    }
  }
  testID="TreeView"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    focusable={true}
    layout={[Function]}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexDirection": "row",
        "gap": 4,
      }
    }
    testID="TreeViewItem"
  >
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "aspectRatio": 1,
          "backgroundColor": "beige",
          "borderRadius": 8,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 20,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="black"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        >
          <RNSVGPath
            d="M5 12h14"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M12 5v14"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      layout={[Function]}
      style={
        {
          "flex": 1,
          "gap": 8,
          "overflow": "hidden",
        }
      }
      testID="TreeViewDataRoot"
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "lineHeight": 20,
            },
            false,
          ]
        }
      >
        Phone Category
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    focusable={true}
    layout={[Function]}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexDirection": "row",
        "gap": 4,
      }
    }
    testID="TreeViewItem"
  >
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "aspectRatio": 1,
          "backgroundColor": "beige",
          "borderRadius": 8,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 20,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="black"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        >
          <RNSVGPath
            d="M5 12h14"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M12 5v14"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      layout={[Function]}
      style={
        {
          "flex": 1,
          "gap": 8,
          "overflow": "hidden",
        }
      }
      testID="TreeViewDataRoot"
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "lineHeight": 20,
            },
            false,
          ]
        }
      >
        Laptop Category
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    focusable={true}
    layout={[Function]}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexDirection": "row",
        "gap": 4,
      }
    }
    testID="TreeViewItem"
  >
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "aspectRatio": 1,
          "backgroundColor": "beige",
          "borderRadius": 8,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 20,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="black"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        >
          <RNSVGPath
            d="M5 12h14"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M12 5v14"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      layout={[Function]}
      style={
        {
          "flex": 1,
          "gap": 8,
          "overflow": "hidden",
        }
      }
      testID="TreeViewDataRoot"
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "lineHeight": 20,
            },
            false,
          ]
        }
      >
        Tablet Category
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    focusable={true}
    layout={[Function]}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexDirection": "row",
        "gap": 4,
      }
    }
    testID="TreeViewItem"
  >
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "aspectRatio": 1,
          "backgroundColor": "beige",
          "borderRadius": 8,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 20,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="black"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        >
          <RNSVGPath
            d="M5 12h14"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M12 5v14"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      layout={[Function]}
      style={
        {
          "flex": 1,
          "gap": 8,
          "overflow": "hidden",
        }
      }
      testID="TreeViewDataRoot"
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "lineHeight": 20,
            },
            false,
          ]
        }
      >
        Watch Category
      </Text>
    </View>
  </View>
</View>
`;
